genie-agent/src/main/java/com/netflix/genie/agent/cli/CommandsRegistry.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.cli;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Set;
+
+class CommandsRegistry {
```

Comment:
This seems like it might be hard to scale? I think we need to talk about what "commands" we're trying to support in here before we really decide how much we want to support.

Reply:
I don't think we'd have too many... `execute`, the rest could be things for debugging, maybe some porcelain to check on job status. Can't think of much else.

Unless we want to consolidate agent and client (in which case we'd need commands to submit jobs, create/delete/update configurations).

Either way, each new command requires 2 lines in this file, a name and a registration. Why is this not scalable?
Would you like to do a scan for commands rather than hardcoding this list?

Reply:
Can't we just scan for implementations of the Command interface as beans? We can also make them part of a BeanFactory to give us the correct one based on a string similar to what we do with the file download services in Genie web that @ajoymajumdar did